buildenv/jenkins/omrbuild.groovy

```diff
@@ -301,103 +325,304 @@ def SPECS = [
+                            junit '**/*results.xml'
+                        }
+                    }
+                    break
+                case 'autoconf':
+                    sh "make test ${SPECS[buildSpec].testArgs} ${testAppend}"
+                    break
+                default:
+                    error("Unknown buildSystem type")
+            }
+        } else {
+            echo "Currently no sanity tests..."
+        }
+    }
+}
+
+def cleanDockerContainers() {
+    println("Listing docker containers to attempt removal")
+    sh "docker ps -a"
+    def containers = sh (script: "docker ps -a --format \"{{.ID}}\"", returnStdout: true).trim()
+    if (containers) {
+        println("Stop all docker containers")
+        sh "docker ps -a --format \"{{.ID}}\" | xargs docker stop"
+        println("Remove all docker containers")
+        sh "docker ps -a --format \"{{.ID}}\" | xargs docker rm --force"
+    }
+}
+
+def buildDockerImage() {
+    stage("Docker Build") {
+        getSources()
+        dir("buildenv/docker/${arch}/${os}") {
+            dockerImage = docker.build(dockerImageName)
+        }
+    }
+}
+
+def runAllStages() {
+    /* Use a custom workspace name.
+     * This is becasue the LIBPATH on z/os includes the build's workspace.
+     * Since we cannot add a variable to the 'environment' in the SPECS, as
+     * it will get evaluated early, we'll use a custom ws in order to
+     * hardcode the value in the SPECS.
+     */
+    def customWorkspace = WORKSPACE - "/${JOB_NAME}" + "/${workspaceName}"
+    ws(customWorkspace) {
+        try {
+            timeout(time: 2, unit: 'HOURS') {
+                def tmpDesc = (currentBuild.description) ? currentBuild.description + "<br>" : ""
+                currentBuild.description = tmpDesc + "<a href=${JENKINS_URL}computer/${NODE_NAME}>${NODE_NAME}</a>"
+
+                def environmentVars = SPECS[buildSpec].environment
+
+                if (runInDocker) {
+                    environmentVars += "OMR_RUNNING_IN_DOCKER=1"
+                }
+
+                if (!envAppend.isEmpty()) {
+                    environmentVars.addAll(envAppend)
+                }
+
+                println "env vars: " + environmentVars
+
+                withEnv(environmentVars) {
+                    sh 'printenv'
+                    getSources()
+                    build();
+                    test();
+                }
+            }
+        } finally {
+            if (!pullId && scmVars) {
+                setBuildStatus("Complete", currentBuild.currentResult, "${scmVars.GIT_COMMIT}")
+            }
+            cleanWs()
+        }
+    }
+}
+
+def runAllStagesInDocker() {
+    def customWorkspace = WORKSPACE - "/${JOB_NAME}" + "/${workspaceName}_docker"
+    ws(customWorkspace) {
+        try {
+            cleanDockerContainers()
+            buildDockerImage()
+            dockerImage.inside("-v /home/jenkins:/home/jenkins:rw,z") {
```

Comment:
Do we need the entire home/jenkins folder mounted in the container? I think we use it for the repo cache?

Reply:
In the `SPECS` data struct, there are multiple references to `/home/jenkins/*` for Linux specifications. Mounting the entire `/home/jenkins` will allow us to support those specs from within a container.